pkg/authenticator/password/ldappassword/ldap.go

```diff
@@ -42,6 +43,7 @@ type Authenticator struct {
 
 // New returns an authenticator which will validate usernames/passwords using LDAP.
 func New(providerName string, options Options, mapper authapi.UserIdentityMapper) (authenticator.Password, error) {
+	ldap.DefaultTimeout = 30 * time.Second
```

Comment:
Please add a comment about why you're doing this and that it is a global variable of the ldap module that we're writing into.

Reply:
addressed it! please take a look!

Reply:
So this means that out of 60 sec request timeout we have 30s left at worst? This still kills every request taking longer than 30s to be processed by etcd.

Reply:
Yes, I agree. There are other failure cases too. If there are 3 IDPs configured(assuming it is a realistic scenario) and if the first two IDPs take 30s, request would be killed without trying the third IDP. 

I am not sure what repercussions increasing the request timeout would have. I didn't feel comfortable increasing it and so i ended up reducing the ldap timeout. Do you think it is better or there is a better way of solving it?